WcaOnRails/app/controllers/posts_controller.rb

```diff
@@ -5,7 +5,13 @@ class PostsController < ApplicationController
   before_action -> { redirect_to_root_unless_user(:can_create_posts?) }, except: [:index, :rss, :show]
 
   def index
-    @posts = Post.where(world_readable: true).order(sticky: :desc, created_at: :desc).includes(:author).page(params[:page])
+    tag = params[:tag]
+    if tag
+      @posts = Post.joins(:post_tags).where('post_tags.tag = ?', tag)
+    else
+      @posts = Post.joins("LEFT JOIN post_tags ON post_tags.post_id = posts.id").group("posts.id").having("IFNULL(SUM(tag = 'wdc'), 0) = 0")
```

Comment:
> But why aren't they removed in the other if branch?

They aren't removed in the other if branch because we want to show posts tagged `wdc` if the user asks for all posts tagged `wdc`. 

> Do they only appear as the only tag?

Sorry, I'm not sure what this is asking. Currently no posts have any tags. I don't know how future posts will be tagged. I know you've been reading the "revamp our homepage" email thread, if one thing is clear, it's that nobody knows what the future holds for us yet =)

> Maybe this should really be a boolean column instead of a tag?

That would certainly make this query easier. One idea would be to add a column titled `show on homepage` that defaults to true, but users can intentionally unselect. That was actually my original plan, but I decided to go with this because we're discussing changing the homepage and "show on homepage" might lose its meaning. Now I'm starting to think that an explicit "show on homepage" boolean column might be best. We can always remove it in the future if we want to.